src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1897,6 +1988,100 @@ private long EnforceConfigurationFor(DocumentsOperationContext context, string i
             }
         }
 
+        private bool AdoptOrphanedFor(DocumentsOperationContext context, string id)
+        {
+            using (DocumentIdWorker.GetSliceFromId(context, id, out var lowerId))
+            using (GetKeyPrefix(context, lowerId, out var lowerIdPrefix))
+            {
+                var collectionName = GetCollectionFor(context, lowerIdPrefix);
+                if (collectionName == null)
+                {
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"Tried to delete revisions for '{id}' but no revisions found.");
+                    return false;
+                }
+
+
+                if (ShouldCreateDeleteRevision(context, lowerId, lowerIdPrefix, collectionName, out var table, out var lastRevision))
+                {
+                    var lastModifiedTicks = _database.Time.GetUtcNow().Ticks;
+                    var newEtag = _database.DocumentsStorage.GenerateNextEtag();
+                    var cv = _documentsStorage.GetNewChangeVector(context, newEtag);
+                    CreateDeleteRevision(context, table, id, collectionName, cv, lastModifiedTicks, lastRevision.Flags);
+                    return true;
+                }
+
+                return false;
+            }
+        }
+
+        internal void DeleteLastRevisionFor(DocumentsOperationContext context, string id)
+        {
+            var collectionName = new CollectionName("Users");
+            using (DocumentIdWorker.GetSliceFromId(context, id, out var lowerId))
+            using (GetKeyPrefix(context, lowerId, out var lowerIdPrefix))
+            using (GetKeyWithEtag(context, lowerId, etag: long.MaxValue, out var compoundPrefix))
+            {
+                var table = EnsureRevisionTableCreated(context.Transaction.InnerTransaction, collectionName);
+                var holder = table.SeekOneBackwardFrom(RevisionsStorage.RevisionsSchema.Indexes[RevisionsStorage.IdAndEtagSlice], lowerIdPrefix, compoundPrefix);
+                var lastRevision = TableValueToRevision(context, ref holder.Reader, DocumentFields.ChangeVector | DocumentFields.LowerId);
+                DeleteRevisionFromTable(context, table, new Dictionary<string, Table>(), lastRevision, collectionName, lastRevision.ChangeVector, _database.Time.GetUtcNow().Ticks);
+                IncrementCountOfRevisions(context, lowerIdPrefix, -1);
+            }
+
+        }
+
+        private unsafe void CreateDeleteRevision(DocumentsOperationContext context, Table table, string id, CollectionName collectionName, 
```

Comment:
why not to use the `Delete` method?

Reply:
https://github.com/ravendb/ravendb/pull/17355#discussion_r1342095426